test/interactor/MemberInteractorTest.java

```diff
@@ -0,0 +1,62 @@
+package interactor;
+
+import entity.MemberTask;
+import entity.Profile;
+import entity.Team;
+import gateway.IGateway;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import java.util.List;
+
+public class MemberInteractorTest {
+
+    private String email;
+    private MemberInteractor memberInteractor;
+
+    private class FakeProjectStateManager implements IGateway {
+        @Override
+        public Profile getProfile(String e) { 
+            Profile p = new Profile();
+            p.email = e;
+            return p;
+        }
+        @Override
+        public boolean isFirstProfile(){ return false; }
+        @Override
+        public void saveProfile(Profile p){}
+        @Override
+        public void saveTeam(Team t){}
+        @Override
+        public boolean isUniqueTeamName(String n){ return true; }
+        @Override
+        public int getNumTeams(){return 1;}
+        @Override
+        public List<Team> getOpenTeams() { return null; }
+        @Override
+        public List<Profile> getProfiles(Team t) { return null; }
+        @Override
+        public MemberTask getMemberTask(String e) { return null; }
+    }
+
+
+
+
+    @Before
+    public void setup() {
+        memberInteractor = new MemberInteractor(new FakeProjectStateManager());
+        email = "test@gmail.com";
+    }
+
+    @Test
```

Comment:
```suggestion
@Test
public void instanceOfIMember(){
    assertTrue( memberInteractor instanceof IMember);
}

    @Test
```

Reply:
This test should have drove you to implement the IMember interface and all its functions inside MemberInteractor.

Reply:
I know I didn't have these sorts of tests in my code, but I went ahead and put them in as they are good documentation showing what the interactor implements and also an alert if the implementation is removed.